.github/workflows/node.js.yml

```diff
@@ -0,0 +1,41 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Dola Flash Minter CI
+
+env:
+  COVERAGE_GIT_BRANCH: "master"
+  INFURA_API_KEY: ${{ secrets.INFURA_API_KEY }}
+  MNEMONIC: ${{ secrets.MNEMONIC }}
+  ETHERSCAN_API_KEY: ${{ secrets.ETHERSCAN_API_KEY }}
+  ALCHEMY_TOKEN: ${{ secrets.ALCHEMY_TOKEN }}
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: "Install dependencies"
+        run: "yarn install --frozen-lockfile"
+
+      - name: "Compile the contracts and generate TypeChain artifacts"
+        run: "yarn typechain"
+
```

Comment:
Don't forget to run the lint job

Reply:
I've removed the linting step on purpose to stay in scope with this PR (add testing). We'll open a new one to clean things up, including pre-existing code.